run_matching.R

```diff
@@ -0,0 +1,194 @@
+  if (!length(path_own_sector_classification) == 1) {
+    stop("When matching_use_own_sector_classification == TRUE, argument path_own_sector_classification must be of length 1. Please check your input.")
+  }
+  if (!inherits(path_own_sector_classification, "character")) {
+    stop("When matching_use_own_sector_classification == TRUE, argument path_own_sector_classification must be of class character. Please check your input.")
+  }
+}
+
+# load data----
+
+## load abcd----
+if (!file.exists(path_abcd)) {
+  stop(glue::glue("No ABCD file found at path {path_abcd}. Please check your project setup!"))
+}
+
+abcd <- readr::read_csv(
+  file.path(path_abcd),
+  col_types = col_types_abcd,
+  col_select = dplyr::all_of(col_select_abcd)
+)
+
+## optionally load own classification system----
+if (matching_use_own_sector_classification) {
+  if (!file.exists(path_own_sector_classification)) {
+    stop(glue::glue("No sector classification file found at path {path_own_sector_classification}. Please check your project setup!"))
+  }
+
+  sector_classification_system <- readr::read_csv(
+    file = path_own_sector_classification,
+    col_types = readr::cols_only(
+      code_system = "c",
+      code = "c",
+      sector = "c",
+      borderline = "l"
+    )
+  ) %>%
+    dplyr::select(names(r2dii.data::sector_classifications))
+}
+
+# load, match, and save loan books----
+list_raw <- list.files(dir_raw)[grepl(".csv", list.files(dir_raw))]
+
+if (length(list_raw) == 0) {
+  stop(glue::glue("No raw loan book csvs found in {dir_raw}. Please check your project setup!"))
+}
+
+for (i in list_raw) {
+  # load raw loan books
+  raw_lbk_i <- readr::read_csv(
+    file.path(dir_raw, i),
+    col_types = col_types_raw
+  )
+
+  group_name <- gsub(".csv", "", i)
+
+  # match data
+  if (matching_use_own_sector_classification) {
+    withr::with_options(
+      new = list(r2dii.match.sector_classifications = sector_classification_system),
+      code = {
+        getOption("r2dii.match.sector_classifications")
+        matched_lbk_i <- r2dii.match::match_name(
+          loanbook = raw_lbk_i,
+          abcd = abcd,
+          by_sector = matching_by_sector,
+          min_score = matching_min_score,
+          method = matching_method,
+          p = matching_p,
+          overwrite = matching_overwrite,
+          join_id = matching_join_id
+          # TODO: allow surfacing the other match_name args
+        )
+      }
+    )
+  } else {
+    matched_lbk_i <- r2dii.match::match_name(
+      loanbook = raw_lbk_i,
+      abcd = abcd,
+      by_sector = matching_by_sector,
+      min_score = matching_min_score,
+      method = matching_method,
+      p = matching_p,
+      overwrite = matching_overwrite,
+      join_id = matching_join_id
+    )
+  }
+
+  # write matched data to file
+  matched_lbk_i %>%
+    readr::write_csv(
+      file = file.path(dir_matched, glue::glue("matched_lbk_{group_name}.csv")),
+      na = ""
+    )
```

Comment:
NIT: In general, do you think it's preferable to write out X number of `.csv` per matched loanbook? or 1 single `matched.csv` object, with an `id` col indicating the loanbook?

Reply:
I think it is better to have X separate `.csv` files, because I think it's easier to handle separate smaller files in the manual matching step. also makes splitting manual work between people a bit easier to handle